app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -1102,6 +1172,74 @@ private void showInfoLayout() {
         return result;
     }
 
+    private void initOCR() {
+        String extdir = getExternalFilesDir(null).toString();
+        if (!new File(extdir + "/tessdata/eng.traineddata").exists()) {
+            copyAssetFolder(getAssets(), "tessdata", extdir + "/tessdata");
+        }
+
+        ocr = OCRHelper.init(extdir, displayMetrics.widthPixels, displayMetrics.heightPixels);
+        ocr.nidoFemale = getResources().getString(R.string.pokemon029);
+        ocr.nidoMale = getResources().getString(R.string.pokemon032);
+    }
+
+
+
+    /**
+     * scanPokemon
+     * Performs OCR on an image of a pokemon and sends the pulled info to PokeFly to display.
+     *
+     * @param pokemonImage The image of the pokemon
+     * @param filePath     The screenshot path if it is a file, used to delete once checked
+     */
+    private void scanPokemon(Bitmap pokemonImage, String filePath) {
+        //WARNING: this method *must* always send an intent at the end, no matter what, to avoid the application hanging.
+        Intent info = Pokefly.createNoInfoIntent();
+        if (ocr == null) {
+            Toast.makeText(Pokefly.this, "Screen analysis module not initialized", Toast.LENGTH_LONG).show();
```

Comment:
Either the above warning is no more needed (because of this PR), or this path crashes the app. Not that you introduced it, but maybe you fixed it?